internal/rbd/types/volume.go

```diff
@@ -25,9 +25,48 @@ import (
 	"github.com/ceph/ceph-csi/internal/util"
 )
 
-//nolint:interfacebloat // more than 10 methods are needed for the interface
+type snapshottableVolume interface {
+	// NewSnapshotByID creates a new Snapshot object based on the details of the Volume.
+	NewSnapshotByID(ctx context.Context, cr *util.Credentials, name string, id uint64) (Snapshot, error)
+
+	// PrepareVolumeForSnapshot prepares the volume for snapshot by
+	// checking snapshots limit and clone depth limit and flatten it
+	// if required.
+	PrepareVolumeForSnapshot(ctx context.Context, cr *util.Credentials) error
+
+	// HandleParentImageExistence checks the image's parent.
+	// if the parent image does not exist and is not in trash, it returns nil.
+	// if the flattenMode is FlattenModeForce, it flattens the image itself.
+	// if the parent image is in trash, it returns an error.
+	// if the parent image exists and is not enabled for mirroring, it returns an error.
+	HandleParentImageExistence(ctx context.Context, flattenMode FlattenMode) error
```

Comment:
This is required for csi addons replication enable operation & group add operation only and not for CSI spec calls.